specification.md

```diff
@@ -2474,6 +2531,207 @@ authenticity of the downloaded file, for example can be the
 The exact signing and verification method is Agent specific. See
 [Code Signing](#code-signing) for recommendations.
 
+### Custom Messages
+
+#### Motivation
+
+The OpAMP protocol is intended to cover all aspects of remote management and configuration
+of Agents. If more requirements within this core functionality are identified, it is
+expected that the protocol will be extended to support this additional behavior in future
+releases. However, there are some use cases that require communication with Agents that
+will not by supported by the OpAMP protocol because they are not within the scope of Agent
+remote management. CustomCapabilities and CustomMessage allow custom behavior to be
+implemented between a Server and Agent without having to open another connection or define
+an entirely new protocol.
+
+While interoperability between many Agents and Agent Management Servers is a goal of
+OpAMP, additional custom functionality may only be supported by specific Agents and
+Servers. Where appropriate, implementers are encouraged to document their use of
+CustomMessage so that support can be added to additional platforms. If appropriate, custom
+capabilities that are widely supported by Server and Agent implementations may be adopted
+as standard OpAMP capabilities in the future.
+
+#### CustomCapabilities
+
+The Agent and Server should both use this message to signal that they support specific
+custom capabilities. It is supported on ServerToAgent and AgentToServer. When this message
+is sent, the list of supported custom capabilities should be updated to match this list.
+If this message is never sent, it is assumed that no custom message types are supported.
+
+Capabilities are identified by a reverse FQDN with optional version information. For
+example, "com.company.capability/v2" identifies version 2 of "capability" created by
```

Comment:
Is there a defined format for version number, e.g. semver?

Reply:
This isn't used at all by the protocol but we wanted to encourage implementers to follow a standard convention. I think semver is fine and we could document it as such.